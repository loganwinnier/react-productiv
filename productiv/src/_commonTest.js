const TEST_TODO = {
  id: 1,
  title: "todo",
  description: "desc",
  priority: 1 }

const TEST_TODOS = [
  {
  id: 1,
  title: "todo-1",
  description: "desc-1",
  priority: 1
  },
  {
    id: 2,
    title: "todo-2",
    description: "desc-2",
    priority: 2
  },
  {
    id: 3,
    title: "todo-3",
    description: "desc-3",
    priority: 3
  }]

export { TEST_TODO, TEST_TODOS }
