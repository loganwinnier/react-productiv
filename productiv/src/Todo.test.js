import React from "react";
import { fireEvent, render } from "@testing-library/react";
import Todo from "./Todo";
import { TEST_TODO } from "./_commonTest";

describe("Todo component", function () {
  it("renders without crashing", function () {
    render(<Todo todo={TEST_TODO}/>);
  });

  it("contains title, description, priority", function () {
    const result = render(<Todo todo={TEST_TODO}/>);
    expect(result.queryByText(TEST_TODO.title)).toBeInTheDocument();
    expect(result.queryByText(TEST_TODO.description)).toBeInTheDocument();
    expect(result.queryByText(TEST_TODO.priority)).toBeInTheDocument();
  });

  it ("matches snapshot", function () {
    const { container } = render(<Todo todo={TEST_TODO}/>);
    expect(container).toMatchSnapshot();
  })
});