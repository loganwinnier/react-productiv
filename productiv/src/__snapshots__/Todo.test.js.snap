// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo component matches snapshot 1`] = `
<div>
  <div
    class="Todo"
  >
    <div>
      <b>
        todo
      </b>
       
      <small>
        1
      </small>
    </div>
    <div>
      <small>
        desc
      </small>
    </div>
  </div>
</div>
`;
